plugins/pulp_rpm/plugins/migrations/0043_populate_recommends.py

```diff
@@ -0,0 +1,74 @@
+import gzip
+import logging
+import sys
+
+from pulp.server.db.connection import get_collection
+from pulp.server.db.migrations.lib import utils
+
+if sys.version_info < (2, 7):
+    from xml.etree import ElementTree as ET
+else:
+    from xml.etree import cElementTree as ET
+
+
+_logger = logging.getLogger(__name__)
+
+
+_NAMESPACES = {
+    'common': "http://linux.duke.edu/metadata/common",
+    'rpm': 'http://linux.duke.edu/metadata/rpm',
+}
+
+_HEADER = ('<?mxl version="1.0" encoding="UTF-8"?>\n'
+           '<metadata xmlns="http://linux.duke.edu/metadata/common" '
+           'xmlns:rpm="http://linux.duke.edu/metadata/rpm">\n {}'
+           '</metadata>\n')
+
+
+def migrate_rpm(collection, unit):
+    """
+    Uncompress single rmp unit primary metadata and populate the unit recommends attribute
+    accordingly.
+
+    :param collection: a collection of RPM units
+    :type collection: pymongo.collection.Collection
+    :param unit: the RPM unit being migrated
+    :type unit: dict
+    """
+    primary_xml = unit.get('repodata', {}).get('primary')
+    if not primary_xml:
+        _logger.warning('Encountered an RPM unit without the primary repodata: %s', unit['_id'])
```

Comment:
i would remove this logging, this message does not bring anything useful to the user and might just confuse or alert him.